main.tf

```diff
@@ -100,16 +114,17 @@ resource "ibm_is_vpc" "vpc" {
 resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_id" {
   count = (var.enable_hub == false && var.enable_hub_vpc_id) ? 1 : 0
 
-  name   = "${var.prefix}-dns-binding"
+  name   = var.dns_binding_name != null ? var.dns_binding_name : (var.prefix != null ? "${var.prefix}-dns-binding" : "${var.name}-dns-binding")
   vpc_id = local.vpc_id # Source VPC
   vpc {
     id = var.hub_vpc_id # Target VPC ID
   }
 }
 
 resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_crn" {
-  count  = (var.enable_hub == false && var.enable_hub_vpc_crn) ? 1 : 0
-  name   = "${var.prefix}-dns-binding"
+  count = (var.enable_hub == false && var.enable_hub_vpc_crn) ? 1 : 0
+
+  name   = var.dns_binding_name != null ? var.dns_binding_name : (var.prefix != null ? "${var.prefix}-dns-binding" : "${var.name}-dns-binding")
```

Comment:
Using coalesce may make this easier to read. Also I'd suggest adding a comment above.